modules/dnn/src/layers/fast_convolution/fast_convolution.hpp

```diff
@@ -21,7 +21,7 @@ enum { FAST_VEC_NLANES=4 };
 #define CONV_MR 4
 #define CONV_NR 24
 
-#ifdef CV_AVX2
+#if CV_AVX2
```

Comment:
Thanks for the reminder, what macros should we use to distinguish function calls? How about `#if __AVX2__`?

Reply:
According to comment `// SIMD 128` in the else branch you don't want to handle AVX2 here only (because code is a priori broken for other SIMD256 ISAs).
I believe you want to check for SIMD256 / 128.
So you should do that through CV_SIMD_WIDTH check in that case.

Reply:
I got your point now. For now, we have AVX2 branch, NEON branch, and Universal intrinsics (SIMD 128).
And if we just set `FAST_VEC_NLANES to 8` when SIMD256 is true. We need add the then Universal intrinsics (SIMD 256) implementation to prevent code errors.

Reply:
I will reconsider how to better support more platforms.

Reply:
Hi @alalek, the current AVX implementation is compatible with AVX and AVX2. How can the implementation of a function (like `convBlock_AVX2`) exist in two namespaces (`opt_AVX` and `opt_AVX2`) at the same time?

Reply:
Updata: The current workaround is that AVX2 will be computed at AVX2 branch, and AVX to the SIMD256 branch.

Reply:
Thank you for update!
I would propose to keep original one-line "quick fix" to unlock win32/linux32 builds and then move refactoring into separate PR (as there is still open questions).

---

> How can the implementation of a function (like convBlock_AVX2) exist in two namespaces (opt_AVX and opt_AVX2) at the same time?

This should be implemented with "runtime dispatching" through `.simd.hpp` files (finally we should not have SIMD code in .cpp files). See also this wiki page: https://github.com/opencv/opencv/wiki/CPU-optimizations-build-options

Reply:
Agree.